ruby.rb

```diff
@@ -0,0 +1,146 @@
+#  #####          Q1 answer      ########
+
+ students = [
+    {
+        first_name: 'John',
+        last_name: 'Doe'
+    },
+    {
+        first_name: 'Jane',
+        last_name: 'Doe',
+    },
+    {
+        first_name: 'Jennifer',
+        last_name: 'Aniston',
+    }
+  ]
+  
+  upper_case_full_names = students.map do |x|
+    x[:first_name].upcase+" "+ x[:last_name].upcase
+  end
+  p upper_case_full_names
+ 
+
+
+
+#####          Q2 answer      ########
+
+
+
+users = [
+  {
+      name: 'Ahmed',
+      orders: [
+          {
+              description: 'a bike'
+          }
+      ]
+  },
+  {
+      name: 'Sami',
+      orders: [
+          {
+              description: 'bees'
+          },
+          {
+              description: 'fishing rod'
+          }
+      ]
+  },
+  {
+      name: 'Salman',
+      orders: [
+          {
+              description: 'a MacBook'
+          },
+          {
+              description: 'The West Wing DVDs'
+          },
+          {
+              description: 'headphones'
+          },
+          {
+              description: 'a kitten'
+          }
+      ]
+  }
+]
+
+first_order_for_each_user = users.map do |i|
```

Comment:
Great work, but the variable's name represents what information the variable contains so change the variable name from ```i``` to ```student```
``` students.each do |student|```